### PR TITLE
feat(Combobox): allow custom end content

### PR DESCRIPTION
**Pull request by elliot-wilson**
Design would like to have a Combobox that doesn't have a up/down chevron indicating whether the Combobox is open/closed.

To address that in a more extensible way, I propose letting the consumer supply a `renderEndContent` function that receives an `isOpen` prop.

image

### CODE REVIEW

File: src/Combobox/index.js [hunk 1]
```diff
@@ -363,11 +364,12 @@ const Combobox = ({
           value={inputValue}
           startIcon={icon}
           endContent={
+            inputEndContent === undefined ? (
```
Comment (akdetrick) [inline src/Combobox/index.js:367]: I was going to suggest a prop default, but I see the scope of the render is needed to decide the icon

**Review by akdetrick — COMMENTED** [review]
@elliot-wilson Do you think we'll ever need to make the end content aware of the open state?

If so, we should do this as a function instead:

`endContent={(isOpen) => <h1>this is {isOpen ? "open" : "closed"}, lol</h1>}`


Love the approach here either way

**Comment by elliot-wilson** [comment]
> @elliot-wilson Do you think we'll ever need to make the end content aware of the open state?
> 
> If so, we should do this as a function instead:
> 
> `endContent={(isOpen) => <h1>this is {isOpen ? "open" : "closed"}, lol</h1>}`
> 
> Love the approach here either way

oh, yes, you're exactly right -- a function is the way to go. I'll update this!

File: src/Combobox/index.js [hunk 2]
```diff
@@ -121,6 +135,7 @@ const Combobox = ({
   errorText,
   icon,
   testId,
+  renderEndContent = defaultRenderEndContent,
```
Comment (akdetrick) [inline src/Combobox/index.js:138]: **Optional**

Renaming this `endContent` would make it easier to remember (although types will help in the future anyway)

Comment (akdetrick) [reply src/Combobox/index.js:138]: oh, that's why:

```
* Signature: `(isOpen) => React.ReactNode`
```

**Review by akdetrick — APPROVED** [review]

**Pull request merged**